custom_components/lutron_caseta_pro/fan.py

```diff
@@ -149,13 +149,13 @@ async def async_turn_off(self, **kwargs) -> None:
     def update_state(self, value):
         """Update internal state and fan speed."""
         self._is_on = value > SPEED_MAPPING[SPEED_OFF]
-        if SPEED_MAPPING[SPEED_MEDIUM_HIGH] < value <= SPEED_MAPPING[SPEED_HIGH]:
+        if value == SPEED_MAPPING[SPEED_HIGH]:
```

Comment:
What is the reason for dropping the support for ranges here? The Lutron Integration Protocol specifies the speeds as supported ranges and not exact values.

Reply:
My switch is actually in an inconvenient spot so I rarely use it. I can double check what telnet says with a physical button press. The values I found reported via telnet when using HomeKit and the Lutron app. Since they were whole numbers, I assumed the range was not needed anymore as it worked in my config both setting and reporting the state.

Reply:
Okay, with the way it is setup right now, values like 75.0% from Homekit should fall within the correct range and should be producing the correct state in Home Assistant.

For reference, here is how Lutron lists the speed values in their protocol document:
```
Levels:
0 =Off
1–25%=Low
26–50%=Medium
56–75%=Medium High
76–100%=High
```

Reply:
<img width="770" alt="Screen Shot 2020-12-20 at 10 55 54 AM" src="https://user-images.githubusercontent.com/35356217/102719204-602b6280-42b2-11eb-8733-9efdb2a441e2.png">

I changed the speed to all 5 settings (Off, Low, Medium, Medium High, High) in the order of HomeKit, Home Assistant, Physical Fan Control, and finally Lutron App.  These are the values output to telnet.  If we follow the ranges Lutron lists in their protocol, then it should work and we can leave the value/range checking in place.  However, as I noted in my issue, the states are off by one such that setting the fan to low in HA actually sets the fan to medium.  It is impossible to set the fan to low in HA.  Can only be set to low physically, from the Lutron app, or from HomeKit.

Reply:
I can confirm these values on my system as well. The new reported values through the physical buttons are:
```
Off: 0.00
Low: 25.00
Medium: 50.00
Medium-High: 75.00
High: 100.00
```

I also confirmed that updating the values in `fan.py` resolved the control issue and Home Assistant can properly set all of the fan speeds again.